dispatcherd/service/process.py

```diff
@@ -107,10 +118,20 @@ async def read_finished(self) -> dict[str, Union[str, int]]:
         message = await self.get_event_loop().run_in_executor(None, self.finished_queue.get)
         return message
 
+    def shutdown(self) -> None:
+        if self._finished_queue:
+            logger.debug('Closing finished queue')
```

Comment:
Would be nice to know exactly which worker is emitting this log message.

Reply:
Oh... I guess that's the non-obvious thing. There is only one finished queue.